tests/integration/test_ha.py

```diff
@@ -243,3 +260,88 @@ async def test_cluster_preserves_data_on_delete(ops_test: OpsTest) -> None:
             select_data_sql,
         )
         assert random_chars in output
+
+
+@pytest.mark.order(5)
+@pytest.mark.ha_tests
+async def test_cluster_isolation(ops_test: OpsTest) -> None:
+    """Test for cluster data isolation.
+
+    This test creates a new cluster, create a new table on both cluster, write a single record with
+    the application name for each cluster, retrieve and compare these records, asserting they are
+    not the same.
+    """
+    app = await app_name(ops_test)
+    apps = [app, ANOTHER_APP_NAME]
+
+    # Build and deploy secondary charm
+    charm = another_charm or await ops_test.build_charm(".")
```

Comment:
Is there a situation where you need to build here again, instead of just using `another_charm`?